### ISSUE TITLE
Implementation with MvvmCross

### ISSUE THREAD

**Issue by meilleur102** [FantasticFiasco/mvvm-dialogs#147]
Hi! It's more a question than a bug. 
Im pretty new to MVVM and I struggle for the Dialog implementation. 

The MvvmDialog add on seems really nice but I can't figure how to implement it with MvvmCross. 
I check the Custom Dialog type Locator. Create my Locator class but I can't find the way to have the Mvx.IoC working. 

Somebody know ho to do it? 
The IoC is the confusing par for me. I don't understand what class or interface I need to put where on the IoC.
Here's the link for the MvvmCross Dependency injection 
https://www.mvvmcross.com/documentation/fundamentals/dependency-injection

If you need more info let me know. But for now I just want to have a Modal for a yes no question or a alert with a OK button. 
Thanks. 
Jc

**Comment by FantasticFiasco** [comment]
Hi!

I'll see if I can build a sample application using MvvmCross. It would be a good complement to the sample applications written for other MVVM frameworks.

**Comment by meilleur102** [comment]
> 
> 
> Hi!
> 
> I'll see if I can build a sample application using MvvmCross. It would be a good complement to the sample applications written for other MVVM frameworks.

That will be awesome. 
The MVVMCross with the core and the wpf not on the same project make it really hard to implement some new plugins.

**Comment by FantasticFiasco** [comment]
Can you point me to a repo where they show off really simple application built using MVVMCross for WPF?

**Comment by meilleur102** [comment]
> 
> 
> Can you point me to a repo where they show off really simple application built using MVVMCross for WPF?

Ill build you a starter project 2night and fwd it to you

**Comment by meilleur102** [comment]
> 
> 
> > Can you point me to a repo where they show off really simple application built using MVVMCross for WPF?
> 
> Ill build you a starter project 2night and fwd it to you

This is the link for the repo 
https://github.com/meilleur102/MvvmCross-Sample

Ive put some comment on the RootViewModel for you to understand how the navigation work for this sample. 
I did not install the MvvmDialog via the Nuget I wanted you to have a core project. 
Let me know if you need more info. 
Thanks. 
Jc

**Comment by FantasticFiasco** [comment]
Thanks! I'll see if I can get it to work with this library.

**Comment by meilleur102** [comment]
> 
> 
> Thanks! I'll see if I can get it to work with this library.

Exelent thx. Let me know.

**Comment by FantasticFiasco** [comment]
I need a clarification. In `RootViewModel` you want to open a new dialog, I'm with you on that. But would you like to use `IDialogService` from this library there instead of `IMvxNavigationService`, or do you still want to use `IMvxNavigationService`?

Because, coming from a place where I know nothing of MvvmCross, it doesn't seem wise to reference `MvvmDialogs` from `MvxBase.Core` since `MvvmDialogs` only targets WPF and UWP. I don't even know what would happen if you tried run the code on another platform. Can you please clarify what we are trying to accomplish?

**Comment by meilleur102** [comment]
The core project it's only because it can be multiplatform. But on my case it's wpf only. 
I just use the convention of MVVCross. 2 Project. One with the WPF and the core.

The goal here is to be able to have a Dialog window, file explorer etc. Ive found MvvmDialog can do all that really easy. 
The IMvxNavigationService that Ive put it's only to show you how this framework work. You don't have to use it if you don't need to.

**Comment by meilleur102** [comment]
Clarification*****
You can remove the ImvxNavicationService from the RootViewModel constructor and implement the IDialog there that will change nothing to the function of the application.. The IMvxCommand for the button click will still work with IDialog. Its really the convertion from a MvxViewModel to a MvvmDialogViewModel the problem. I think the IDialog is looking for some info about the view that he can't retreive because it's not on the same project.

**Comment by FantasticFiasco** [comment]
Ok, thanks for the clarification. I'll continue with the attempt tonight.

**Comment by meilleur102** [comment]
> 
> 
> Ok, thanks for the clarification. I'll continue with the attempt tonight.

good thx!

**Comment by FantasticFiasco** [comment]
I've created a pull request in your sample repo.

**Comment by meilleur102** [comment]
awesome thanks

**Comment by meilleur102** [comment]
I just did some text and it work perfectly. I try the file explorer and work as well. The only thing that don't work is the MessageBox. I try to implement the IMessageBox but Ive got a error
The type is defined in an assembly that is not referenced. You must add a reference to assembly.
Ill just do the messagebox with the DialogWindow and that it! 
Thanks again for your help! 
I really struggle with the implementation. Still a junior!

**Comment by FantasticFiasco** [comment]
No problem, I hope your workaround with the message box is sufficient. I don't think MvvmCross will be officially supported by this library, but if it lets you be more productive you should continue using it.

Best of luck in the future!

**Comment by meilleur102** [comment]
> 
> 
> No problem, I hope your workaround with the message box is sufficient. I don't think MvvmCross will be officially supported by this library, but if it lets you be more productive you should continue using it.
> 
> Best of luck in the future!

Thx a lot again. One last question. What mvvm framework you recommend using with MvvmDialog? 
Im not 100% on MvvmCross yet so still time to change

**Comment by FantasticFiasco** [comment]
Well I would use MvvmCross if I wanted to target multiple platforms, but the applications I've been a part of always only targeted Windows. And usually we'd start with MVVM Light or something similar, but we'd always down the line replace the parts for other components.

As an example, many of the MVVM frameworks come with a IoC container built in, and that will get you a long way, but at some point you will replace it with a "real" IoC container like e.g. [Autofac](https://autofac.org/). And most MVVM frameworks let you do that, they often support injecting your own IoC container.

And then perhaps you don't like the event dispatcher that came with the MVVM framework, and again you replace that one.

In the end, after replacing bit by bit, you reduce the value of the MVVM framework. At that point you perhaps only rely on the viewmodel base classes, and those are easy to write yourself. So you write those, and you eject out of the MVVM framework, rolling with your own implementation. You are now better off. You've gained a lot of knowledge along the way, and none of the parts you depend on are irreplaceable. You now "own" your infrastructure.

Long story short, it doesn't really matter where you start, because you will change along the way, and you shouldn't put to much effort into choosing the correct framework up front, because most probably you don't know what's good for **you**. Not all developers think alike, you'll gravitate towards what makes sense to you.

**Comment by meilleur102** [comment]
> 
> 
> Well I would use MvvmCross if I wanted to target multiple platforms, but the applications I've been a part of always only targeted Windows. And usually we'd start with MVVM Light or something similar, but we'd always down the line replace the parts for other components.
> 
> As an example, many of the MVVM frameworks come with a IoC container built in, and that will get you a long way, but at some point you will replace it with a "real" IoC container like e.g. [Autofac](https://autofac.org/). And most MVVM frameworks let you do that, they often support injecting your own IoC container.
> 
> And then perhaps you don't like the event dispatcher that came with the MVVM framework, and again you replace that one.
> 
> In the end, after replacing bit by bit, you reduce the value of the MVVM framework. At that point you perhaps only rely on the viewmodel base classes, and those are easy to write yourself. So you write those, and you eject out of the MVVM framework, rolling with your own implementation. You are now better off. You've gained a lot of knowledge along the way, and none of the parts you depend on are irreplaceable. You now "own" your infrastructure.
> 
> Long story short, it doesn't really matter where you start, because you will change along the way, and you shouldn't put to much effort into choosing the correct framework up front, because most probably you don't know what's good for **you**. Not all developers think alike, you'll gravitate towards what makes sense to you.

Cool thanks for the info!
